interfaces/builtin/kernel_module.go

```diff
@@ -0,0 +1,48 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"github.com/snapcore/snapd/interfaces"
+)
+
+var kernelModuleConnectedPlugAppArmor = `
+# Description: Allow insertion of modules.
+
+  capability sys_module,
+`
+
+var kernelModuleConnectedPlugSecComp = `
+# Description: Allow insertion of modules.
+
+finit_module
+delete_module
+`
+
+// NewKernelModuleInterface returns a new "kernel-module" interface.
+func NewKernelModuleInterface() interfaces.Interface {
+	return &commonInterface{
+		name: "kernel-module",
+		connectedPlugAppArmor: kernelModuleConnectedPlugAppArmor,
+		connectedPlugSecComp:  kernelModuleConnectedPlugSecComp,
+		reservedForOS:         true,
+		autoConnect:           true,
```

Comment:
If an application snap is allowed to use the 'kernel-module' interface, is there a way to connect this interface from within the application? Or will that require an additional manual step?

I'm hoping one could just install the application snap and it be able to insert modules as appropriate.